codice/ddf
# DDF 4033 2.13.x Removing the default jetty error page by injecting in a better one

#### What does this PR do?
Adds a default mapping of error pages to jetty to avoid stack traces being put into the browser by injecting /ErrorServlet into all the registered bundles containing a ServletContext

#### Who is reviewing it? 
Who is reviewing it?
@blen-desta @brjeter @bakejeyner 

Select relevant component teams:
https://github.com/orgs/codice/teams

Ask 2 committers to review/merge the PR and tag them here.
@coyotesqrl
@stustison
@clockard 

#### How should this be tested?


1) Sabotage ddf to cause a consistent jetty error
  I did this by commenting out WEBSSOFilter and having it throw a new runtime exception [like so](https://github.com/brjeter/ddf/commit/b4d8f901e3a74ea09674372c271a68157cec8ac8#diff-367bcf0277415a8dc68979cd35f37738)

2) Build and start DDF.

3) Stop the error page injector bundle in Karaf (`list | grep -i Injector`)

4) Install a profile ( this should create new servlets that can be injected into) and start the error page injector. The logs should have information about error pages being injected

5) Recreate the error that was implemented you should see a default error message to view the logs
and no stacktrace

#### Any background context you want to provide?
#### What are the relevant tickets?
[DDF-4033](https://codice.atlassian.net/browse/4033)
[DDF-4063](https://codice.atlassian.net/browse/4063)
#### Screenshots

#### Checklist:

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
